### PR TITLE
perf(linter): use cow_utils in no_script_url

### PR DESCRIPTION
**Pull request by heygsc**
part of #5586 

I don't know if this is right or not🤣

### CODE REVIEW

File: crates/oxc_linter/Cargo.toml [hunk 1]
```diff
@@ -36,6 +36,7 @@ oxc_syntax = { workspace = true }
 
 bitflags = { workspace = true }
 convert_case = { workspace = true }
+cow-utils = { workspace = true }
```
Comment (DonIsaac) [inline crates/oxc_linter/Cargo.toml:39]: Please run `taplo fmt`, I think our CI skips this.

**Review by DonIsaac — COMMENTED** [review]
This is a good start! Im sure other PRs will update other parts of the code.

File: crates/oxc_linter/Cargo.toml (hunk 1 above)
Comment (Boshen) [reply crates/oxc_linter/Cargo.toml:39]: It's not skipped anymore!

**Comment by Boshen** [comment]
This is great! Keep going! You may find inefficient routines while working on this ;-)

File: crates/oxc_linter/Cargo.toml (hunk 1 above)
Comment (heygsc) [reply crates/oxc_linter/Cargo.toml:39]: I ran `dprint fmt` locally again and didn't see any changes, so do I need to make changes here?

Comment (Boshen) [reply crates/oxc_linter/Cargo.toml:39]: No it's not needed.

**Comment by heygsc** [comment]
Thank you both. I didn't make any further changes because I didn't know if what I did was correct at the time.

Since it seems right now, I will try to do more.

**Comment by heygsc** [comment]
Will this be merged? Or should I continue to modify other rules in this PR?

**Review by camc314 — APPROVED** [review]

**Comment by shulaoda** [comment]
> Open a new PR? Or should I continue to modify other rules here?

Opening a new PR is welcome.

**Pull request merged**